src/cli.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{CliArgs, Client, GenesisDetails, StateId};
+use ethereum_consensus::{phase0::mainnet::Fork, primitives::Root};
+use url::Url;
+
+pub async fn cli(args: CliArgs) {
+    //set up client
+    let s = args.node_url;
+    let url: Url = Url::parse(&s).unwrap();
+    let client = Client::new(url);
+    let endpoint = args.endpoint.as_str();
+
+    match endpoint {
+        // Match on the provided endpoint
+        "node_version" => println!("{:?}", client.get_node_version().await.unwrap()),
+        "genesis" => {
+            let out: GenesisDetails = client.get_genesis_details().await.unwrap();
+            println!("{:?}", out.genesis_time);
+            println!("{:?}", out.genesis_fork_version);
+            println!("{:?}", out.genesis_validators_root);
+        }
+        "root" => {
+            if let Some(state_id) = args.state_id {
+                let state = parse_state_id(state_id);
+                let out: Root = client.get_state_root(state).await.unwrap();
+                println!("{out:?}");
+            } else {
+                println!("no state-id provided");
+            }
+        }
+        "fork" => {
+            if let Some(state_id) = args.state_id {
+                let state = parse_state_id(state_id);
+                let out: Fork = client.get_fork(state).await.unwrap();
```

Comment:
question: I would expect this to work *without* the explicit `Fork` type annotation on `out`-- does it?